Getting ClassNotFoundException for GrpcSslContexts

HI,

I am trying to use this jar to publish events. When I'm trying to connect a client, I'm getting the following exception.
`Caused by: java.lang.ClassNotFoundException: io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts`


My code :

```
private static Client client;

void init() {
    ClientSettings setts = ConnectionString.parseOrThrow("esdb://my_ip:2113");
    client = Client.create(setts);
}
```

StackTrace:
```
io.netty.channel.ChannelPipelineException: io.micronaut.http.netty.stream.HttpStreamsHandler$2.handlerRemoved() has thrown an exception.
        at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:640) ~[eventstore-0.1.jar:?]
        at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:477) ~[eventstore-0.1.jar:?]
        at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:423) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.stream.HttpStreamsHandler.removeHandlerIfActive(HttpStreamsHandler.java:429) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.stream.HttpStreamsHandler.handleReadHttpContent(HttpStreamsHandler.java:294) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.stream.HttpStreamsHandler.channelRead(HttpStreamsHandler.java:257) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.stream.HttpStreamsServerHandler.channelRead(HttpStreamsServerHandler.java:121) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[eventstore-0.1.jar:?]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[eventstore-0.1.jar:?]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) ~[eventstore-0.1.jar:?]
        at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[eventstore-0.1.jar:?]
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93) ~[eventstore-0.1.jar:?]
        at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) ~[eventstore-0.1.jar:?]
        at io.netty.handler.flow.FlowControlHandler.dequeue(FlowControlHandler.java:200) ~[eventstore-0.1.jar:?]
        at io.netty.handler.flow.FlowControlHandler.read(FlowControlHandler.java:139) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:686) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:671) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher.requestDemand(HandlerPublisher.java:163) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.stream.HttpStreamsHandler$2.requestDemand(HttpStreamsHandler.java:248) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.receivedDemand(HandlerPublisher.java:547) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.lambda$request$0(HandlerPublisher.java:474) ~[eventstore-0.1.jar:?]
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [eventstore-0.1.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [eventstore-0.1.jar:?]
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497) [eventstore-0.1.jar:?]
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [eventstore-0.1.jar:?]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [eventstore-0.1.jar:?]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [eventstore-0.1.jar:?]
        at java.lang.Thread.run(Thread.java:830) [?:?]
Caused by: java.lang.NoClassDefFoundError: io/grpc/netty/shaded/io/grpc/netty/GrpcSslContexts
        at com.eventstore.dbclient.Client.<init>(Client.java:23) ~[eventstore-0.1.jar:?]
        at com.eventstore.dbclient.Client.create(Client.java:52) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.util.EventStoreUtil.init(EventStoreUtil.java:25) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.util.EventStoreUtil.publishSingleEvent(EventStoreUtil.java:58) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produceEvent(ProducerServiceImpl.java:42) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.lambda$produce$0(ProducerServiceImpl.java:29) ~[eventstore-0.1.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1507) ~[?:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produce(ProducerServiceImpl.java:29) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.controller.ProducerController.produce(ProducerController.java:25) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.controller.$ProducerControllerDefinition$$exec1.invokeInternal(Unknown Source) ~[eventstore-0.1.jar:?]
        at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146) ~[eventstore-0.1.jar:?]
        at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:474) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:312) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$10(RoutingInBoundHandler.java:1365) ~[eventstore-0.1.jar:?]
        at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14865) ~[eventstore-0.1.jar:?]
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14868) ~[eventstore-0.1.jar:?]
        at io.micronaut.tracing.instrument.util.TracingPublisher.subscribe(TracingPublisher.java:119) ~[eventstore-0.1.jar:?]
        at io.micronaut.configuration.metrics.binder.web.WebMetricsPublisher.subscribe(WebMetricsPublisher.java:155) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.context.ServerRequestContextFilter.lambda$doFilter$0(ServerRequestContextFilter.java:62) ~[eventstore-0.1.jar:?]
        at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14865) ~[eventstore-0.1.jar:?]
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14868) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildExecutableRoute$6(RoutingInBoundHandler.java:1070) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.DefaultUriRouteMatch$1.execute(DefaultUriRouteMatch.java:80) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler$2.doOnComplete(RoutingInBoundHandler.java:996) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnComplete(JsonContentProcessor.java:140) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at java.util.Optional.ifPresent(Optional.java:176) ~[?:?]
        at io.micronaut.core.async.processor.SingleThreadedBufferingProcessor.doOnComplete(SingleThreadedBufferingProcessor.java:48) ~[eventstore-0.1.jar:?]
        at io.micronaut.jackson.parser.JacksonProcessor.doOnComplete(JacksonProcessor.java:137) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.SingleThreadedBufferingSubscriber.onComplete(SingleThreadedBufferingSubscriber.java:70) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.jackson.JsonContentProcessor.doOnComplete(JsonContentProcessor.java:165) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher.complete(HandlerPublisher.java:416) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher.handlerRemoved(HandlerPublisher.java:403) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:946) ~[eventstore-0.1.jar:?]
        at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637) ~[eventstore-0.1.jar:?]
        ... 38 more
Caused by: java.lang.ClassNotFoundException: io.grpc.netty.shaded.io.grpc.netty.GrpcSslContexts
        at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602) ~[?:?]
        at jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
        at com.eventstore.dbclient.Client.<init>(Client.java:23) ~[eventstore-0.1.jar:?]
        at com.eventstore.dbclient.Client.create(Client.java:52) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.util.EventStoreUtil.init(EventStoreUtil.java:25) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.util.EventStoreUtil.publishSingleEvent(EventStoreUtil.java:58) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produceEvent(ProducerServiceImpl.java:42) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.lambda$produce$0(ProducerServiceImpl.java:29) ~[eventstore-0.1.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1507) ~[?:?]
        at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produce(ProducerServiceImpl.java:29) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.controller.ProducerController.produce(ProducerController.java:25) ~[eventstore-0.1.jar:?]
        at com.xxxxx.eventstore.controller.$ProducerControllerDefinition$$exec1.invokeInternal(Unknown Source) ~[eventstore-0.1.jar:?]
        at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146) ~[eventstore-0.1.jar:?]
        at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:474) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:312) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$10(RoutingInBoundHandler.java:1365) ~[eventstore-0.1.jar:?]
        at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14865) ~[eventstore-0.1.jar:?]
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14868) ~[eventstore-0.1.jar:?]
        at io.micronaut.tracing.instrument.util.TracingPublisher.subscribe(TracingPublisher.java:119) ~[eventstore-0.1.jar:?]
        at io.micronaut.configuration.metrics.binder.web.WebMetricsPublisher.subscribe(WebMetricsPublisher.java:155) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.context.ServerRequestContextFilter.lambda$doFilter$0(ServerRequestContextFilter.java:62) ~[eventstore-0.1.jar:?]
        at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14865) ~[eventstore-0.1.jar:?]
        at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14918) ~[eventstore-0.1.jar:?]
        at io.reactivex.Flowable.subscribe(Flowable.java:14868) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildExecutableRoute$6(RoutingInBoundHandler.java:1070) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.DefaultUriRouteMatch$1.execute(DefaultUriRouteMatch.java:80) ~[eventstore-0.1.jar:?]
        at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.RoutingInBoundHandler$2.doOnComplete(RoutingInBoundHandler.java:996) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnComplete(JsonContentProcessor.java:140) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at java.util.Optional.ifPresent(Optional.java:176) ~[?:?]
        at io.micronaut.core.async.processor.SingleThreadedBufferingProcessor.doOnComplete(SingleThreadedBufferingProcessor.java:48) ~[eventstore-0.1.jar:?]
        at io.micronaut.jackson.parser.JacksonProcessor.doOnComplete(JacksonProcessor.java:137) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.SingleThreadedBufferingSubscriber.onComplete(SingleThreadedBufferingSubscriber.java:70) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.server.netty.jackson.JsonContentProcessor.doOnComplete(JsonContentProcessor.java:165) ~[eventstore-0.1.jar:?]
        at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher.complete(HandlerPublisher.java:416) ~[eventstore-0.1.jar:?]
        at io.micronaut.http.netty.reactive.HandlerPublisher.handlerRemoved(HandlerPublisher.java:403) ~[eventstore-0.1.jar:?]
        at io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:946) ~[eventstore-0.1.jar:?]
        at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637) ~[eventstore-0.1.jar:?]
        ... 38 more
```

I'm using version **0.2**

```
<dependency>
    <groupId>com.eventstore</groupId>
    <artifactId>db-client-java</artifactId>
    <version>0.2</version>
</dependency>
```

@YoEight I'm using Java 11. On adding the jar, that error is resolved.  But I'm getting a new error.

`Caused by: java.lang.ClassNotFoundException: io.grpc.stub.AbstractStub`

Yes. Exact same.  That was the first thing I tried which lead me to build it manually.

On adding the dependencies you mentioned above, I think its going past that.

I had to add 

```<!-- https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java -->
        <dependency>
            <groupId>com.google.protobuf</groupId>
            <artifactId>protobuf-java</artifactId>
            <version>4.0.0-rc-2</version>
        </dependency>
        <!-- https://mvnrepository.com/artifact/io.grpc/grpc-stub -->
        <dependency>
            <groupId>io.grpc</groupId>
            <artifactId>grpc-stub</artifactId>
            <version>1.33.1</version>
        </dependency>

        <!-- https://mvnrepository.com/artifact/io.grpc/grpc-protobuf -->
        <dependency>
            <groupId>io.grpc</groupId>
            <artifactId>grpc-protobuf</artifactId>
            <version>1.33.1</version>
        </dependency>

        <!-- https://mvnrepository.com/artifact/com.fasterxml.jackson.core/jackson-databind -->
        <dependency>
            <groupId>com.fasterxml.jackson.core</groupId>
            <artifactId>jackson-databind</artifactId>
            <version>2.11.3</version>
        </dependency>

        <dependency>
            <groupId>io.grpc</groupId>
            <artifactId>grpc-netty-shaded</artifactId>
            <version>1.33.1</version>
        </dependency>
```

Now I'm getting a different error.  May or may not be related. 

```
java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2070)
	at com.xxxxx.eventstore.util.EventStoreUtil.publishSingleEvent(EventStoreUtil.java:67)
	at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produceEvent(ProducerServiceImpl.java:42)
	at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.lambda$produce$0(ProducerServiceImpl.java:29)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1507)
	at com.xxxxx.eventstore.service.impl.ProducerServiceImpl.produce(ProducerServiceImpl.java:29)
	at com.xxxxx.eventstore.controller.ProducerController.produce(ProducerController.java:25)
	at com.xxxxx.eventstore.controller.$ProducerControllerDefinition$$exec1.invokeInternal(Unknown Source)
	at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:146)
	at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:474)
	at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:312)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildResultEmitter$10(RoutingInBoundHandler.java:1365)
	at io.reactivex.internal.operators.flowable.FlowableDefer.subscribeActual(FlowableDefer.java:35)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14868)
	at io.micronaut.tracing.instrument.util.TracingPublisher.subscribe(TracingPublisher.java:119)
	at io.micronaut.configuration.metrics.binder.web.WebMetricsPublisher.subscribe(WebMetricsPublisher.java:155)
	at io.micronaut.http.server.context.ServerRequestContextFilter.lambda$doFilter$0(ServerRequestContextFilter.java:62)
	at io.reactivex.internal.operators.flowable.FlowableFromPublisher.subscribeActual(FlowableFromPublisher.java:29)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.micronaut.reactive.rxjava2.RxInstrumentedFlowable.subscribeActual(RxInstrumentedFlowable.java:57)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14868)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.lambda$buildExecutableRoute$6(RoutingInBoundHandler.java:1070)
	at io.micronaut.web.router.DefaultUriRouteMatch$1.execute(DefaultUriRouteMatch.java:80)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:118)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$2.doOnComplete(RoutingInBoundHandler.java:996)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnComplete(JsonContentProcessor.java:140)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at java.base/java.util.Optional.ifPresent(Optional.java:176)
	at io.micronaut.core.async.processor.SingleThreadedBufferingProcessor.doOnComplete(SingleThreadedBufferingProcessor.java:48)
	at io.micronaut.jackson.parser.JacksonProcessor.doOnComplete(JacksonProcessor.java:137)
	at io.micronaut.core.async.subscriber.SingleThreadedBufferingSubscriber.onComplete(SingleThreadedBufferingSubscriber.java:70)
	at io.micronaut.http.server.netty.jackson.JsonContentProcessor.doOnComplete(JsonContentProcessor.java:165)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onComplete(CompletionAwareSubscriber.java:71)
	at io.micronaut.http.netty.reactive.HandlerPublisher.complete(HandlerPublisher.java:416)
	at io.micronaut.http.netty.reactive.HandlerPublisher.handlerRemoved(HandlerPublisher.java:403)
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerRemoved(AbstractChannelHandlerContext.java:946)
	at io.netty.channel.DefaultChannelPipeline.callHandlerRemoved0(DefaultChannelPipeline.java:637)
	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:477)
	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:423)
	at io.micronaut.http.netty.stream.HttpStreamsHandler.removeHandlerIfActive(HttpStreamsHandler.java:429)
	at io.micronaut.http.netty.stream.HttpStreamsHandler.handleReadHttpContent(HttpStreamsHandler.java:294)
	at io.micronaut.http.netty.stream.HttpStreamsHandler.channelRead(HttpStreamsHandler.java:257)
	at io.micronaut.http.netty.stream.HttpStreamsServerHandler.channelRead(HttpStreamsServerHandler.java:121)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.flow.FlowControlHandler.dequeue(FlowControlHandler.java:200)
	at io.netty.handler.flow.FlowControlHandler.read(FlowControlHandler.java:139)
	at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:686)
	at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:671)
	at io.micronaut.http.netty.reactive.HandlerPublisher.requestDemand(HandlerPublisher.java:163)
	at io.micronaut.http.netty.stream.HttpStreamsHandler$2.requestDemand(HttpStreamsHandler.java:248)
	at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.receivedDemand(HandlerPublisher.java:547)
	at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.lambda$request$0(HandlerPublisher.java:474)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute$$$capture(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
Channel Pipeline: [SslHandler#0, ProtocolNegotiators$ClientTlsHandler#0, WriteBufferingAndExceptionHandler#0, DefaultChannelPipeline$TailContext#0]
	at io.grpc.Status.asRuntimeException(Status.java:533)
	at io.grpc.stub.ClientCalls$StreamObserverToCallListenerAdapter.onClose(ClientCalls.java:478)
	at io.grpc.internal.DelayedClientCall$DelayedListener$3.run(DelayedClientCall.java:464)
	at io.grpc.internal.DelayedClientCall$DelayedListener.delayOrExecute(DelayedClientCall.java:428)
	at io.grpc.internal.DelayedClientCall$DelayedListener.onClose(DelayedClientCall.java:461)
	at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:617)
	at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:70)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:803)
	at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:782)
	at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
	at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	... 1 more
Caused by: io.grpc.netty.shaded.io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record: 485454502f312e31203430302042616420526571756573740d0a436f6e6e656374696f6e3a20636c6f73650d0a446174653a204672692c203237204e6f7620323032302031373a35363a343520474d540d0a5365727665723a204b65737472656c0d0a436f6e74656e742d4c656e6774683a20300d0a0d0a
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1254)
	at io.grpc.netty.shaded.io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1322)
	at io.grpc.netty.shaded.io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:501)
	at io.grpc.netty.shaded.io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:440)
	at io.grpc.netty.shaded.io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.grpc.netty.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.grpc.netty.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.grpc.netty.shaded.io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.grpc.netty.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.grpc.netty.shaded.io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.grpc.netty.shaded.io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.grpc.netty.shaded.io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163)
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:714)
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.grpc.netty.shaded.io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.grpc.netty.shaded.io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.grpc.netty.shaded.io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.grpc.netty.shaded.io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	... 1 more

```

--- a/db-client-java/build.gradle
+++ b/db-client-java/build.gradle
@@ -19,15 +19,15 @@ group = 'com.eventstore'
 version = '0.2'
 
 dependencies {
-    implementation "javax.annotation:javax.annotation-api:${annotationApiVersion}"
-    implementation "javax.validation:validation-api:${validationApiVersion}"
+    api "javax.annotation:javax.annotation-api:${annotationApiVersion}"
+    api "javax.validation:validation-api:${validationApiVersion}"
 
-    implementation "com.google.protobuf:protobuf-java:${protobufVersion}"
-    implementation "io.grpc:grpc-netty-shaded:${grpcVersion}"
-    implementation "io.grpc:grpc-stub:${grpcVersion}"
-    implementation "io.grpc:grpc-protobuf:${grpcVersion}"
-	implementation 'dnsjava:dnsjava:3.2.2'
-    implementation 'com.fasterxml.jackson.core:jackson-databind:2.11.1'
+    api "com.google.protobuf:protobuf-java:${protobufVersion}"
+    api "io.grpc:grpc-netty-shaded:${grpcVersion}"
+    api "io.grpc:grpc-stub:${grpcVersion}"
+    api "io.grpc:grpc-protobuf:${grpcVersion}"
+    api 'dnsjava:dnsjava:3.2.2'
+    api 'com.fasterxml.jackson.core:jackson-databind:2.11.1'
 
     testImplementation "junit:junit:${junitVersion}"
     testImplementation "org.slf4j:slf4j-nop:${slf4jNopVersion}"

